lib/requests.js

```diff
@@ -76,55 +120,55 @@ function _getDataAndStatus({result = {}, error = {}}) {
   return {data, statusCode};
 }
 
-function _deleteAuthorizationHeader({error}) {
+function _deleteErrorHeaders({error}) {
   if(error.response) {
-    const {headers: responseHeaders} = error.response;
-    // Clone the response headers
-    const newHeaders = new global.Headers(responseHeaders);
-    // delete the Authorization header to prevent
-    // oauth2 headers potentially in logs
-    newHeaders.delete('Authorization');
-    const newResponse = new global.Response(error.response, {
-      headers: newHeaders
+    error.response = _deleteResponseHeaders({response: error.response});
+  }
+  if(error.request) {
+    error.request = new global.Request(error.request, {
+      headers: _deleteHeaders({httpMessage: error.request})
     });
-    error.response = newResponse;
   }
   return error;
 }
 
+function _deleteResponseHeaders({response}) {
+  const newResponse = new global.Response(JSON.stringify(response.data), {
+    headers: _deleteHeaders({httpMessage: response}),
+    status: response.status,
+    statusText: response.statusText
+  });
+  if(response.data) {
+    // transfer the already parsed data to the newResponse
+    // we are overwriting the data getter in Response here
+    Object.defineProperty(newResponse, 'data', {value: response.data});
+  }
+  return newResponse;
+}
+
 /**
- * Makes an https request.
+ * Takes in either a response or request & sanitizes the headers.
+ *
+ * @private
  *
  * @param {object} options - Options to use.
- * @param {URL} options.url - A url.
- * @param {object} [options.json] - JSON for the request.
- * @param {object} [options.headers] - Headers for the request.
- * @param {string} options.method - The HTTP method for the request.
- * @param {object} [options.oauth2] - OAuth2 credentialss.
- * @param {object} [options.searchParams] - URL Queries for the request.
+ * @param {global.Response|global.Request} options.httpMessage - A http message.
+ * @param {Array<string>} [options.headers=sanitizeHeaders] - A list of headers
+ *   to delete from the http message.
  *
- * @returns {object} The results from the request.
+ * @returns {global.Headers} - Returns http headers.
  */
-export async function makeHttpsRequest({
-  url,
-  json,
-  headers,
-  method,
-  oauth2,
-  searchParams
-}) {
-  let result;
-  let error;
-  if(oauth2) {
-    headers.Authorization = await constructOAuthHeader({...oauth2});
+function _deleteHeaders({httpMessage, headers = sanitizeHeaders}) {
+  if(!httpMessage) {
+    return new global.Headers();
   }
-  try {
-    result = await httpClient(url, {
-      method, json, headers, agent, searchParams
-    });
-  } catch(e) {
-    error = _deleteAuthorizationHeader({error: e});
+  const {headers: messageHeaders} = httpMessage;
+  // Clone the response headers
+  const newHeaders = new global.Headers(messageHeaders);
+  for(const header of headers) {
```

Comment:
Does this work if the "Authorization" header is lowercase?

Reply:
yes it is case insensitive.